File: js-code/number_of_islands_200.js





var numIslands = function(grid) {

    traverse = (x, y) => {
        if(x < 0 || x >= grid.length || y < 0 || y >= grid.length){
            console.log(grid[x][y]);
            grid[x][y] = 0;
            traverse(grid[x][y]);
        }
    }

    for(let x = 0; x < grid.length; x++){
        for(let y = 0; y < grid[0].length; y++){
            // console.log(grid[x][y]);
            traverse(grid[x][y]);
        }
        console.log('\n');
    }



};



const grid = [
    ["1","1","1","1","0"],
    ["1","1","0","1","0"],
    ["1","1","0","0","0"],
    ["0","0","0","0","0"]
];



let soln = numIslands(grid);
// console.log(soln);

File: python-code/Graphs/number_of_islands_200.py
def numIslandsHelper(x, y, grid):
  if x < 0 or x >= len(grid) or y < 0 or y >= len(grid[0]) or grid[x][y] == '0':
    return

  grid[x][y] = '0'

  numIslandsHelper(x + 1, y, grid)
  numIslandsHelper(x - 1, y, grid)
  numIslandsHelper(x, y + 1, grid)
  numIslandsHelper(x, y - 1, grid)

def numIslands(grid):
  islands = 0

  for x in range(0, len(grid)):
    for y in range(0, len(grid[0])):
      if grid[x][y] == '1':
        numIslandsHelper(x, y, grid)
        islands += 1

  return islands

grid = [
  ["1","1","0","0","0"],
  ["1","1","0","0","0"],
  ["0","0","1","0","0"],
  ["0","0","0","1","1"]
]

numIslands(grid)
